packages/loader/container-loader/src/container.ts

```diff
@@ -932,19 +932,21 @@ export class Container
 				? summaryTree
 				: combineAppAndProtocolSummary(summaryTree, this.captureProtocolSummary());
 
-		// Whether the combined summary tree has been forced on by either the loader option or the monitoring context.
+		// Whether the combined summary tree has been forced on by either the loader option or the monitoring context or supportedFeatures flag by the service.
 		// Even if not forced on via this flag, combined summaries may still be enabled by service policy.
-		const forceEnableSummarizeProtocolTree =
+		const shouldEnableSummarizeProtocolTree =
 			this.mc.config.getBoolean("Fluid.Container.summarizeProtocolTree2") ??
-			options.summarizeProtocolTree;
+			options.summarizeProtocolTree ??
+			(this._deltaManager.connectionManager.supportedFeatures
+				?.enable_single_commit_summary as boolean);
```

Comment:
I don't think you need to cast to boolean, `ContainerStorageAdapter` constructor takes `boolean | undefined`.  If you _did_ need it to be boolean, it's safer to explicitly convert the `undefined` case to `false`  (who knows if some other code would do `if (x === false)` or something surprising like that, and then there'd be a bug)